File: leetcode-javascript/map/lc0136-只出现一次的数字.js
var singleNumber = function(nums) {
    let map = {};

    for (let num of nums) {
        map[num] = (map[num] || 0) + 1;
    }

    for (let key in map) {
        if (map[key] === 1) return key;
    }
};

var singleNumber = function(nums) {
    return nums.reduce((a,b) => a ^ b)
};

File: leetcode-python/Array/LC0136-只出现一次的数字.py
'''
示例 1 ：

输入：nums = [2,2,1]
输出：1
示例 2 ：

输入：nums = [4,1,2,1,2]
输出：4
'''


class Solution:
    def singleNumber(self, nums):
        m = {}
        for v in nums:
            m[v] = m.get(v, 0) + 1
        for k in m:
            if m[k] == 1:
                return k
        return -1

if __name__ == '__main__':
    nums = [4,1,2,1,2]
    s = Solution()
    print(s.singleNumber(nums))